packages/toolpad-core/src/Account/AccountPreview.test.tsx

```diff
@@ -0,0 +1,81 @@
+/// <reference types="@testing-library/jest-dom/vitest" />
+
+/**
+ * @vitest-environment jsdom
+ */
+
+import * as React from 'react';
+import { describe, test, expect, vi } from 'vitest';
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { AccountPreview } from './AccountPreview';
+import { AppProvider } from '../AppProvider';
+
+describe('AccountPreview', () => {
+  const auth = { signIn: vi.fn(), signOut: vi.fn() };
+  const session = {
+    user: { name: 'John Doe', email: 'john@example.com', image: 'https://example.com/avatar.jpg' },
+  };
+
+  test('renders nothing when no session is provided', () => {
+    render(
+      <AppProvider authentication={auth}>
+        <AccountPreview />
+      </AppProvider>,
+    );
+
+    expect(screen.queryByRole('button')).not.toBeInTheDocument();
+  });
+
+  test('displays condensed variant by default', () => {
+    render(
+      <AppProvider authentication={auth} session={session}>
+        <AccountPreview />
+      </AppProvider>,
+    );
+
+    const avatar = screen.getByRole('img', { name: 'John Doe' });
+    expect(avatar).toBeInTheDocument();
+    expect(avatar).toHaveAttribute('src', 'https://example.com/avatar.jpg');
```

Comment:
To make sure it's the condensed version you might have to check that the text isn't there too maybe?